flocker/acceptance/test_environment_variables.py

```diff
@@ -1,280 +0,0 @@
-                            u"mountpoint": MYSQL_VOLUME_MOUNTPOINT,
-                        },
-                    },
-                },
-            }
-
-            self.mysql_application_different_port = thaw(freeze(
-                self.mysql_application).transform(
-                    [u"applications", MYSQL_APPLICATION_NAME, u"ports", 0,
-                     u"external"], MYSQL_EXTERNAL_PORT + 1))
-            flocker_deploy(self, mysql_deployment, self.mysql_application)
-
-        deploying_mysql = getting_nodes.addCallback(deploy_mysql)
-        return deploying_mysql
-
-    def test_deploy(self):
-        """
-        The test setUp deploys MySQL.
-        """
-        d = assert_expected_deployment(self, {
-            self.node_1: set([MYSQL_APPLICATION]),
-            self.node_2: set([]),
-        })
-
-        return d
-
-    def test_moving_mysql(self):
-        """
-        It is possible to move MySQL to a new node.
-        """
-        flocker_deploy(self, self.mysql_deployment_moved,
-                       self.mysql_application)
-
-        asserting_mysql_moved = assert_expected_deployment(self, {
-            self.node_1: set([]),
-            self.node_2: set([MYSQL_APPLICATION]),
-        })
-
-        return asserting_mysql_moved
-
-    def _get_mysql_connection(self, host, port, user, passwd, db=None):
-        """
-        Returns a ``Deferred`` which fires with a PyMySQL connection when one
-        has been created.
-
-        Parameters are passed directly to PyMySQL:
-        https://github.com/PyMySQL/PyMySQL
-
-        Raise any exceptions thrown when failing to connect if they indicate
-        that MySQL has started.
-        """
-        expected_container = container_configuration_response(
-            MYSQL_APPLICATION, self.node_1_uuid)
-        waiting_for_cluster = succeed(self.cluster)
-
-        def got_cluster(cluster):
-            waiting_for_container = cluster.wait_for_container(
-                expected_container
-            )
-
-            def mysql_connect(result):
-                def mysql_can_connect():
-                    try:
-                        return connect(
-                            host=host,
-                            port=port,
-                            user=user,
-                            passwd=passwd,
-                            db=db,
-                        )
-                    except Error as e:
-                        Message.new(
-                            message_type="acceptance:mysql_connect_error",
-                            error=str(e)).write(Logger())
-                        return False
-                dl = loop_until(mysql_can_connect)
-                return dl
-            waiting_for_container.addCallback(mysql_connect)
-            return waiting_for_container
-
-        waiting_for_cluster.addCallback(got_cluster)
-        return waiting_for_cluster
-
-    @require_pymysql
-    def test_environment_variable_used(self):
```

Comment:
https://clusterhq.atlassian.net/browse/FLOC-1888
and 
https://github.com/docker-library/mysql/issues/71

This tests is really the only novel one, and 
https://github.com/ClusterHQ/flocker/blob/master/flocker/acceptance/test_api.py#L84-L135 tests the same thing. (We lose the test for this code path in `flocker-deploy`, but that is probably acceptable)